### ISSUE TITLE
DependencyFactory::getEventDispatcher triggers a deprecation

### ISSUE THREAD

**Issue by VincentLanglet** [doctrine/migrations#1348]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | x.y.z

#### Summary

`DependencyFactory::getEventDispatcher` is implemented this way:
```
return $this->getDependency(EventDispatcher::class, function (): EventDispatcher {
    return new EventDispatcher(
        $this->getConnection(),
        $this->getConnection()->getEventManager(),
    );
});
```
but `Connection::getEventManager` is deprecated by dbal since https://github.com/doctrine/dbal/issues/5784

Not sure how to fix this @greg0ire @derrabus

**Comment by derrabus** [comment]
Deprecate calling `fromConnection()` without passing an event manager?

**Comment by VincentLanglet** [comment]
Seems like `EntityManagerInterface::getEventManager` exists ; I'll try this instead.

**Comment by derrabus** [comment]
> Seems like `EntityManagerInterface::getEventManager` exists ; I'll try this instead.

Yes, but that'll only work if you have an entity manager. You can use migrations with a connection only.

**Comment by stof** [comment]
`fromConnection` should indeed allow passing an EventManager explicitly, using `Connection::getEventManager` only as fallback (triggering a deprecation telling you to pass an EventManager explicitly).
`fromEntityManager` would register the EventManager using the ORM EventManager.

`DependencyFactory::getEventDispatcher` would then use a `DependencyFactory::getEventManager` method reading the configured event manager.

**Comment by VincentLanglet** [comment]
DoctrineCommand is doing `DependencyFactory::fromConnection` does it mean we have to add `EventManager` to the constructor of all doctrine command ?

**Comment by stof** [comment]
To be compatible with CLI commands, we might need a concept of an event manager loader instead of passing directly an event manager.

**Comment by VincentLanglet** [comment]
I feel like this is solved since I use dbal 4.x without any issues.
